### PR TITLE
feature: upgrade go-swagger to 0.17.1

### PR DESCRIPTION
**Pull request by zhuangqh**
### Ⅰ. Describe what this PR did
upgrade go-swagger to 0.17.1

brief changelog of go-swagger 0.17.1
- stricter validation of swagger spec file
- complete and correct type validation
- x-omitempty extension
- general bufix

### Ⅱ. Does this pull request fix one issue?

ready to fix all of the issue which were in relation to api parameters validation.

### Ⅳ. Describe how to verify it
It is compatible with current code but enable to validate types.

### CODE REVIEW

File: apis/swagger.yml [hunk 1]
```diff
@@ -895,18 +895,10 @@ paths:
...
-          description: "checkpoint id"
-          type: "string"
-        - name: "exit"
-          in: "query"
-          description: "exit after checkpoint"
-          type: "bool"
+        - name: "body"
+          in: "body"
+          schema:
+            $ref: "#/definitions/CheckpointCreateOptions"
```
Comment (allencloud) [inline apis/swagger.yml]: Could you do this kind of change in another pull request? Since then swagger.yml refactoring has nothing to do with the swagger upgrade. WDYT? @zhuangqh

Comment (zhuangqh) [reply apis/swagger.yml]: > Could you do this kind of change in another pull request? Since then swagger.yml refactoring has nothing to do with the swagger upgrade. WDYT? @zhuangqh

Actually, this part of code is wrong, which failed to pass the stricter validation. Futhermore, it makes this PR fail to pass the CI. So this part of change should be included in this PR although it looks confused...

File: apis/swagger.yml [hunk 2]
```diff
@@ -939,19 +929,20 @@ paths:
           $ref: "#/responses/404ErrorResponse"
       tags: ["Container"]
 
-  /containers/{id}/checkpoints/{id}:
+  /containers/{id}/checkpoints/{checkpointId}:
```
Comment (zhuangqh) [inline apis/swagger.yml]: error: duplicated parameter `id`

File: apis/swagger.yml (hunk 1 above)
Comment (allencloud) [reply apis/swagger.yml]: Roger that.

**Comment by allencloud** [comment]
It is so good to see we have make swagger play more important role. I will make this move on as soon as possible.

**Comment by allencloud** [comment]
I am afraid that you still need to update `Dockerfile`, since in it there is https://github.com/alibaba/pouch/blob/5ad887d876d930999e19fd2c2e54f37a18e5754b/Dockerfile#L36 :
```
# install swagger 0.12.0
RUN wget --quiet -O /bin/swagger https://github.com/go-swagger/go-swagger/releases/download/0.12.0/swagger_linux_amd64 \
    && chmod +x /bin/swagger
```

Here we need to update the swagger version to 0.17.1 as well. @zhuangqh

**Comment by allencloud** [comment]
LGTM

**Comment by zhuangqh** [comment]
> LGTM
> we will merge this until the swagger in Dockerfile issue has been solved.

has updated @allencloud
